astropy/stats/funcs.py

```diff
@@ -790,3 +792,322 @@ def mad_std(data):
+        against. Will be called with a vector of all values at once.
+    args : list-like, optional
+        Additional arguments to be supplied to cdf.
+
+    Returns
+    -------
+    D : float
+        The raw statistic.
+    fpp : float
+        The probability of a D this large arising with a sample drawn from
+        the distribution whose CDF is cdf.
+
+    Notes
+    -----
+    The Kuiper statistic resembles the Kolmogorov-Smirnov test in that 
+    it is nonparametric and invariant under reparameterizations of the data. 
+    The Kuiper statistic, in addition, is equally sensitive throughout 
+    the domain, and it is also invariant under cyclic permutations (making 
+    it particularly appropriate for analyzing circular data). 
+
+    Returns (D, fpp), where D is the Kuiper D number and fpp is the 
+    probability that a value as large as D would occur if data was 
+    drawn from cdf.
+
+    Warning: The fpp is calculated only approximately, and it can be 
+    as much as 1.5 times the true value.
+
+    Stephens 1970 claims this is more effective than the KS at detecting 
+    changes in the variance of a distribution; the KS is (he claims) more 
+    sensitive at detecting changes in the mean.
+
+    If cdf was obtained from data by fitting, then fpp is not correct and 
+    it will be necessary to do Monte Carlo simulations to interpret D. 
+    D should normally be independent of the shape of CDF.
+
+    """
+
+    # FIXME: doesn't work for distributions that are actually discrete (for example Poisson).
+    data = np.sort(data)
+    cdfv = cdf(data,*args)
+    N = len(data)
+    D = (np.amax(cdfv-np.arange(N)/float(N))
+         + np.amax((np.arange(N)+1)/float(N)-cdfv))
+
+    return D, kuiper_FPP(D,N)
+
+def kuiper_two(data1, data2):
+    """Compute the Kuiper statistic to compare two samples.
+
+    Parameters
+    ----------
+    data1 : array-like
+        The first set of data values.
+    data2 : array-like
+        The second set of data values.
+    
+    Returns
+    -------
+    D : float
+        The raw test statistic.
+    fpp : float
+        The probability of obtaining two samples this different from
+        the same distribution.
+
+    Notes
+    -----
+    Warning: the fpp is quite approximate, especially for small samples.
+
+    """
+    data1, data2 = np.sort(data1), np.sort(data2)
+
+    if len(data2)<len(data1):
+        data1, data2 = data2, data1
+
+    cdfv1 = np.searchsorted(data2, data1)/float(len(data2)) # this could be more efficient
+    cdfv2 = np.searchsorted(data1, data2)/float(len(data1)) # this could be more efficient
+    D = (np.amax(cdfv1-np.arange(len(data1))/float(len(data1))) + 
+            np.amax(cdfv2-np.arange(len(data2))/float(len(data2))))
+
+    Ne = len(data1)*len(data2)/float(len(data1)+len(data2))
+    return D, kuiper_FPP(D, Ne)
+
+
+
+def fold_intervals(intervals):
+    """Fold the weighted intervals to the interval (0,1).
```

Comment:
I don't understand what this function does. "Fold" invokes "fold with the period" which is quite common in lightcurve analysis, but obviously that's not what is meant here, since there is not way to input a period.